akka-actor/src/main/scala/akka/util/LineNumbers.scala

```
@@ -0,0 +1,320 @@
+          put(nextIdx, str)
+          nextIdx += 1
+        case 3 ⇒ // Integer
+          skip(d, 4)
+          nextIdx += 1
+        case 4 ⇒ // Float
+          skip(d, 4)
+          nextIdx += 1
+        case 5 ⇒ // Long
+          skip(d, 8)
+          nextIdx += 2
+        case 6 ⇒ // Double
+          skip(d, 8)
+          nextIdx += 2
+        case 7 ⇒ // Class
+          val other = d.readUnsignedShort()
+          _xref = _xref.updated(nextIdx, other)
+          nextIdx += 1
+        case 8 ⇒ // String
+          skip(d, 2)
+          nextIdx += 1
+        case 9 ⇒ // FieldRef
+          skip(d, 4) // two shorts
+          nextIdx += 1
+        case 10 ⇒ // MethodRef
+          skip(d, 4) // two shorts
+          nextIdx += 1
+        case 11 ⇒ // InterfaceMethodRef
+          skip(d, 4) // two shorts
+          nextIdx += 1
+        case 12 ⇒ // NameAndType
+          skip(d, 4) // two shorts
+          nextIdx += 1
+        case 15 ⇒ // MethodHandle
+          skip(d, 3) // a byte and a short
+          nextIdx += 1
+        case 16 ⇒ // MethodType
+          skip(d, 2)
+          nextIdx += 1
+        case 18 ⇒ // InvokeDynamic
+          skip(d, 4) // two shorts
+          nextIdx += 1
+      }
+
+  }
+
+  private def forObject(obj: AnyRef): Result =
+    getStreamForClass(obj.getClass).orElse(getStreamForLambda(obj)) match {
+      case None                   ⇒ NoSourceInfo
+      case Some((stream, filter)) ⇒ getInfo(stream, filter)
+    }
+
+  private def getInfo(stream: InputStream, filter: Option[String]): Result = {
+    val dis = new DataInputStream(stream)
+
+    try {
+      skipID(dis)
+      skipVersion(dis)
+      implicit val constants = getConstants(dis)
+      dbg(s"  fwd(${constants.fwd.size}) rev(${constants.rev.size}) ${constants.fwd.keys.toList.sorted}")
+      skipClassInfo(dis)
+      skipInterfaceInfo(dis)
+      skipFields(dis)
+      val lines = readMethods(dis, filter)
+      val source = readAttributes(dis)
```

my original plan was to let that bubble up, but it should probably turn such things into NoSourceInfo, you’re right